### DISCUSSION TITLE
-

### DISCUSSION THREAD

**Discussion by Makezu90** [moudey/Shell#315] [Q&A]

**Comment by RubicBG** [comment]
- The 'apps' and 'Windows' taskbar menus come with the program - the easiest way is to delete part of the code in /imports/taskbar.nss (delete+save and reload the program)
- To get further help, it is necessary to know which version of Nilesoft Shell App you are using.
- describe in detail everything individually that you want to achieve

**Comment by RubicBG** [comment]
open file '\imports\goto.nss' find and delete:
'window.is_taskbar||' (let only this remain 'where=sel.count')

open file '\imports\taskbar.nss' find and delete:
'menu(title="Apps" ....' to the end along with '}'
'menu(title=title.windows ....' to the end along with '}'
'item(title=title.taskbar_Settings ...' the entire line
'item(title=title.desktop ....' the entire line

open file '\imports\taskbar.nss' and add at the end of the file:
```
menu(expanded=1 type='taskbar' pos=0) {
     separator
     item(title=sys.ver.name vis='label' )
     item(title='OS build: '+sys.ver vis='label' )
     separator
}
```

don't forget to reload the app or restart Windows Explorer after each change to see if you did everything right

more about system version commands here:
https://nilesoft.org/docs/functions/sys
![Screenshot 2023-08-20 112034](https://github.com/moudey/Shell/assets/127151533/baa23280-74cc-4c93-905e-db20f90e559d)

**Reply by RubicBG** [reply]
- about position: in the example above, I set it to be at the top with 'pos=0'. Nowhere higher (but other logic may apply to the arrangement). In my picture it looks like it's not at the top, but that's because I want other commands to be there. If you want send me '\imports\taskbar.nss' to look at.
- about 22H2: there is no command to retrieve this information and show it in the menu, otherwise I would have provided a solution for this. I can put it as fixed text, just enough to have it. (I have an idea, when I have time to test it I will write, not what you want, but as close as possible)

**Comment by RubicBG** [comment]
Nilesoft Community / Nilesoft Shell Discord Server
https://discord.com/channels/1106387012707168318/

ME @ Discord 
https://discordapp.com/users/760192930215886918

**Comment by RubicBG** [comment]
it matters which file you import first - here's what to do:

open file '\imports\taskbar.nss' and remove the code you added:
```
menu(expanded=1 type='taskbar' pos=0) {
      separator
      item(title=sys.ver.name vis='label' )
      item(title='OS build: '+sys.ver vis='label' )
      separator
}
```
download [winver.zip](https://github.com/moudey/Shell/files/12394722/winver.zip) , extract the file into ''\imports\', open '\shell.nss' and add before 'import 'imports/terminal.nss' '

```
import 'imports/winver.nss'
```
I use another version of the program and the syntax in it is a little different: I can't test it at my place - write back

**Comment by RubicBG** [comment]
open file '\imports\winver.nss' and replace the code with this one:

```
$svg_winver= '<svg width="100" height="100" viewBox="-160 0 512 512">
        <path style="fill:@image.color2" d="M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z"/>
    </svg>'


menu(expanded=1 type='taskbar' pos=0) {
    separator
    item(title=sys.ver.name image=svg_winver cmd='msinfo32.exe')
    item(title='OS build: '+sys.ver image=svg_winver cmd='winver.exe')
    separator
}
```

test and give an opinion

**Reply by RubicBG** [reply]
```
// minimum requirements: version 1.8.37 (debug version 37)
// https://discord.com/channels/1106387012707168318/1112126326665650176/1130167577386172419
// as commands: copy to clipboard / as commands + key SHIFT: open info apps
menu(expanded=1 type='taskbar' pos=0) {
    separator
    item(title=sys.ver.name image=\uE136 cmd=if(keys.shift(), 'msinfo32.exe', clipboard.set(sys.ver.name)))
    item(title='OS build: '+sys.ver image=\uE136 cmd=if(keys.shift(), 'winver.exe', clipboard.set(sys.ver)))
    separator
}
```

it can even be done:
right click on taskbar -> label item
right click on taskbar while holding shift key - command item

**Comment by RubicBG** [comment]
if it refers to image '?' you can replace them with one of these (example: 'image=\uE136') : 
https://nilesoft.org/gallery/glyphs
or add icon code like the previous one (example: 'image=svg_winver')
```
$svg_winver= '<svg width="100" height="100" viewBox="-160 0 512 512">
         <path style="fill:@image.color2" d="M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v2 12.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 2 4 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z"/>
     </svg>
```

still experimenting with the right image :)

**Reply by RubicBG** [reply]
add 
`vis='static'`

**Comment by RubicBG** [comment]
menu(expanded=1 type='taskbar' pos=0) {
    separator
    item(title=sys.ver.name image=\uE136 vis='static')
    item(title='OS build: '+sys.ver image=\uE136 vis='static')
    separator
}
